packages/portal/backend/src-util/TraverseResults.ts

```diff
@@ -0,0 +1,187 @@
+import Log, {LogLevel} from "../../../common/Log";
+import {AutoTestResult} from "../../../common/types/AutoTestTypes";
+import Util from "../../../common/Util";
+
```

Comment:
@rtholmes Does traversing for the highest grade and saving it, if it is inconsistent, something that we should be recommending to faculty? Was there an issue that resulted in the creation of this file?

Reply:
All of the files in `src-util` are useful to keep around for people to see how to write best-practice code that iterates over the data store directly. I don't think anyone would use them out of the box (e.g., even in 310 we modify them every single time), but they do show a way forward for people to use as a guide.

Reply:
That sounds great. Thanks for the info.